.github/workflows/main.yml

```diff
@@ -46,3 +46,26 @@ jobs:
         working-directory: git-sync
         run: |
           make test
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20.6'
+
+      - uses: actions/checkout@v3
+        with:
+          path: git-sync
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          working-directory: git-sync
+          version: v1.53.3
+
+      - name: make lint
```

Comment:
why make lint AND golangci-lint-action ?

Reply:
Ah, I'll add a comment.  It's because the github action is supposed to format the output such that github will then put comments onto the right lines of the PR.  So I added it first.  But generally I think we want to be able to run things outside of github actions, hence I consider it a more friendly first-pass on the "canonical test", which is `make lint`.

Of course we do now (potentially) have two sources of truth, and some linter warnings (the ones from gostaticcheck) will be presented in the less friendly way.  So not sure whether this is a win or not.  I figured we could try it, but I'm flexible.  WDYT?

Reply:
I am fine with `make lint`, but iwhy would they produce different results?

Reply:
Well `make lint` runs two linters, and the github action form for the second linter didn't look as useful.

But my bigger concern is that while I think they do the same thing today, as we add more options and evolve, drift is possible.  If we weren't "wearing two watches", drift wouldn't be possible.

It's probably a balancing act between how helpful the "pretty" warnings are vs the drift risk over time.  And I suspect we can just evaluate that from experience.  Thanks for lgtm :-)